server/src/controllers/song.controllers.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, RequestHandler, Response } from 'express';
+import { Song }  from '../models/song.model';
+import { SongDocument } from '../types/song.types';
+
+const sendResponse = (res: Response, statusCode: number, payload: SongDocument[] | SongDocument | string ) => {
+  return res.status(statusCode).json(payload);
+};
+
+const createSong: RequestHandler = async (req: Request, res: Response): Promise<SongDocument | string> => {
+  const { ...toCreate } = req.body;
+
+  if (Object.keys(toCreate).length > 0) {
+    try {
+      const data = await Song.create(toCreate);
+
+      if (data) {
+        sendResponse(res, 200, data);
+      } else {
+        sendResponse(res, 404, 'Song not created');
+      }
+    } catch (error: any) {
+      sendResponse(res, 500, error.message);
```

Comment:
Woud be better to use `error?.message`. In case the error doesn't exist, or the whole thing will crash lol. Same thing with the other APIs.